Ready for merge Teslium can now Defib!

sheldoniscool opened the pull request:
## About The Pull Request
Inacusiate: Rapidly repairs damage to the patient's ears to cure deafness. (No longer Instant)
Teslium: Can now defib you if you meet standard defib requirements.

NOTE: 2 downvotes were placed before I removed some synthflesh changes.
## Why It's Good For The Game
I think that Inacusiate shouldn't INSTANTLY heal your hears.

I never see people use teslium redundant with Liquid Electricity. Defibs aren't hard to make and this could add some interesting chem setups.

## Changelog
:cl:
add: Added defibrillation to teslium
tweak: Changed + Inacusiate logic no longer instant.
/:cl:

Commit:
Update lighting.dm

SheldonCurtiss committed:
modified:   code/modules/power/lighting.dm

SheldonCurtiss committed:
modified:   beestation.dme
	modified:   code/modules/reagents/chemistry/reagents/medicine_reagents.dm
	modified:   code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm

sheldoniscool commented:
Synthflesh also needs another pass.

St0rmC4st3r commented on code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm:
```diff
@@ -219,12 +219,89 @@
+		var/failed = 0
+
+		if (H.suiciding)
+			failed = 1
```
you really should add a `return` after each of the `failed = 1`

sheldoniscool replied:
Yeah like I said it's pretty jank - I was just trying to rig it up to make sure everything was working. I plan to update to clean it up but have been pounding on the bee code the past few hours so just wanted to throw this up.

But yeah definitely on the to do list.

PowerfulBacon replied:
Just remove failed entirely and replace it with return

sheldoniscool replied:
This was just a rough draft basically - I finialized sticky tapes today - I'll finish this tomorrow.

St0rmC4st3r commented on code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm:
```diff
@@ -219,12 +219,89 @@
 		shock_timer = 0
-		M.electrocute_act(rand(5,20), "Teslium in their body", 1, 1) //Override because it's caused from INSIDE of you
-		playsound(M, "sparks", 50, 1)
+		H.electrocute_act(rand(5,20), "Teslium in their body", 1, 1) //Override because it's caused from INSIDE of you
```
What't the point?

sheldoniscool replied:
Think this just got moved while testing and rigging everything. Like I said need to do another pass over all of it.
Maybe I should of put draft or something. But yeah it technically works, I want it to work a bit different i.e if you die and it's in your system it will still defib.

St0rmC4st3r reviewed (changes requested):
The PR is not bad. I would recommend adding the returns, where needed for optimization reasons.

Sinmero commented:
Advanced first aid kit has synthflesh patches in it. It is an expensive all in one kit so i think you should add some antitox treatment in it.
Actually i think that synthflesh from being geat healing medicine will become something nobody ever uses cause of high instant tox damage. I think its a good idea to add OD threshold to synthflesh instead.

sheldoniscool commented:
> Advanced first aid kit has synthflesh patches in it. It is an expensive all in one kit so i think you should add some antitox treatment in it.
> Actually i think that synthflesh from being geat healing medicine will become something nobody ever uses cause of high instant tox damage. I think its a good idea to add OD threshold to synthflesh instead.

OD threshold wont work on synthflesh due to it being the item used to prep dead bodies for defib.

Dingo-Dongler commented:
> 
> 
> Advanced first aid kit has synthflesh patches in it. It is an expensive all in one kit so i think you should add some antitox treatment in it.
> Actually i think that synthflesh from being geat healing medicine will become something nobody ever uses cause of high instant tox damage. I think its a good idea to add OD threshold to synthflesh instead.

Advanced kits have pentetic acid pills in them

Dingo-Dongler commented:
> 
> 
> > Advanced first aid kit has synthflesh patches in it. It is an expensive all in one kit so i think you should add some antitox treatment in it.
> > Actually i think that synthflesh from being geat healing medicine will become something nobody ever uses cause of high instant tox damage. I think its a good idea to add OD threshold to synthflesh instead.
> 
> OD threshold wont work on synthflesh due to it being the item used to prep dead bodies for defib.

You would still have to deal with the OD once they are revived, since there's no way to purge chems from dead bodies currently.

sheldoniscool commented:
I've removed the synthflesh changes and will do another PR for them since the teslium is the main focus of this.
Please ignore the done thumbs that are due to this.

I've now updated teslium so it will activate in your system if you die with it inside you.
It requires 5u to trigger (at which point it consumes those units)

SheldonCurtiss committed:
modified:   code/datums/components/rot.dm
	modified:   code/modules/reagents/chemistry/reagents/medicine_reagents.dm
	modified:   code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm

TeomanTheGreat commented:
Very epic

sheldoniscool commented:
@PowerfulBacon Can you take a look at this? Any changes I need to make or anything or can this be merged?

TeomanTheGreat commented:
Lol why did ike close this

ike709 commented:
Because he was repobanned for evasion.
